dataline-integrations/singer/postgres/source/src/test-integration/java/io/dataline/integrations/io/dataline/integration_tests/sources/SingerPostgresSourceTest.java

```diff
@@ -0,0 +1,237 @@
+    schema.getTables().forEach(t -> t.getColumns().forEach(c -> c.setSelected(true)));
+
+    StandardSync syncConfig = new StandardSync().withSyncMode(StandardSync.SyncMode.FULL_REFRESH).withSchema(schema);
+    SourceConnectionImplementation sourceImpl =
+        new SourceConnectionImplementation().withConfiguration(Jsons.jsonNode(getDbConfig()));
+
+    StandardTapConfig tapConfig = new StandardTapConfig()
+        .withStandardSync(syncConfig)
+        .withSourceConnectionImplementation(sourceImpl);
+
+    Stream<SingerMessage> singerMessageStream = singerTapFactory.create(tapConfig, jobRoot);
+
+    SingerMessageTracker singerMessageTracker = new SingerMessageTracker();
+    List<SingerMessage> actualMessages = singerMessageStream.peek(singerMessageTracker).collect(Collectors.toList());
+    for (SingerMessage singerMessage : actualMessages) {
+      LOGGER.info("{}", singerMessage);
+    }
+
+    JsonNode expectedMessagesContainer = Jsons.deserialize(MoreResources.readResource("simple_schema_expected_messages.json")).get("messages");
+    List<SingerMessage> expectedMessages = StreamSupport.stream(expectedMessagesContainer.spliterator(), false)
+        .map(msg -> Jsons.deserialize(Jsons.serialize(msg), SingerMessage.class))
+        .collect(Collectors.toList());
+
+    assertMessagesEquivalent(expectedMessages, actualMessages);
+
+    // test incremental: insert a few more records then run another sync
+    // io.dataline.test.utils.PostgreSQLContainerHelper.runSqlScript(MountableFile.forClasspathResource("simple_postgres_update.sql"),
+    // psqlDb);
+    //
+    // StandardSync incrementalStandardSync = new
+    // StandardSync().withSyncMode(StandardSync.SyncMode.APPEND).withSchema(schema).;
+    // StandardTapConfig incrementalTapConfig = new StandardTapConfig()
+    // .withStandardSync(incrementalStandardSync)
+    // .withState(new State().withState(singerMessageTracker.getOutputState().get()))
+    // .withSourceConnectionImplementation(sourceImpl);
+  }
+
+  // @Test
+  public void testIncrementalRead() {
+    // run an initial read, get the state from it
+    // add a few more records to the DB
+    // pass the state into a second read and make sure the messages are what we expect
+    // TODO
+  }
+
+  private void assertMessagesEquivalent(Collection<SingerMessage> expected, Collection<SingerMessage> actual) {
+    for (SingerMessage expectedMessage : expected) {
+      assertTrue(isMessageContained(expectedMessage, actual), expectedMessage + " was not found in actual messages: " + actual);
+    }
+
+  }
+
+  private static boolean isMessageContained(SingerMessage message, Collection<SingerMessage> collection) {
+    for (SingerMessage containedMessage : collection) {
+      if (!message.getType().equals(containedMessage.getType())) {
+        continue;
+      }
+
+      // We might not want to check that all fields are the same to pass a test e.g: time_extracted isn't
```

Comment:
Could we just overwrite certain fields like `time_extracted` and then just compare the whole object

Reply:
done